src/components/textfieldButtons/js/TextfieldButtons.js

```diff
@@ -0,0 +1,153 @@
+                "method": "val",
+                args: ["{change}.value"]
+            },
+            // TODO: This listener is to deal with the issue that, when the input field receives a invalid input such as a string value,
+            // ignore it and populate the field with the previous value.
+            // This is an area in which UX has spilled over into our model configuration, which to some extent we should try to prevent.
+            // Whenever we receive a "change" event or some other similar checkpoint, if these updates occurred any faster, the user would
+            // be infuriated by being unable to type into the field. This situation doesn't occur at the moment because the change event is
+            // only fired when users leave the input feild. At the very least, we need to give a namespace to this listener - unfortunately
+            // the current dataBinding implementation will ignore it. Having this listener here represents an interaction decision rather
+            // than an implementation decision. This issue needs to be revisited.
+            stringValue: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{that}.model.value"]
+            }
+        },
+        listeners: {
+            "onCreate.bindChangeEvt": {
+                "this": "{that}.container",
+                "method": "change",
+                "args": ["{that}.setModel"]
+            }
+        },
+        invokers: {
+            setModel: {
+                changePath: "stringValue",
+                value: "{arguments}.0.target.value"
+            }
+        }
+    });
+
+    fluid.defaults("fluid.button", {
+        gradeNames: ["fluid.viewComponent"],
+        range: {}, // should be used to specify the min, max range e.g. {min: 0, max: 100}
+        invokers: {
+            setModel: {
+                changePath: "value",
+                value: {
+                    expander: {
+                        funcName: "fluid.buttonCalculateValue",
+                        args: ["{that}.options.incrementCoefficient", "{that}.options.buttonOptions.stepMultiplier", "{that}.model.value", "{that}.options.range.min", "{that}.options.range.max"]
+                    }
+                }
+            }
+        },
+        listeners: {
+            "onCreate.bindClickEvt": {
+                "this": "{that}.container",
+                "method": "click",
+                "args": "{that}.setModel"
+            }
+        }
+    });
+
+    fluid.buttonCalculateValue = function (value, stepMultiplier, modelValue, min, max) {
+        modelValue = Math.round(modelValue * stepMultiplier);
+        min = Math.round(min * stepMultiplier);
+        max = Math.round(max * stepMultiplier);
+        if ((modelValue + value) < min) return min/stepMultiplier;
+        else if ((modelValue + value) > max) return max/stepMultiplier;
+        else return (modelValue + value)/stepMultiplier;
```

Comment:
I'm not sure I follow the logic or reason for the logic in this function. For example why redefine the max and min values? Why the need for rounding? What if I want a step that is decimal value. I'm thinking it would be easier to have two methods:

1. A "setValue" method for setting a specific value
2. A "step" method which performs a step

Also if the button cares about a max and min value, it should also use a model relay to ensure that the value stays within range.